src/main/java/org/jenkinsci/plugins/codescene/CodeSceneBuilder.java

```diff
@@ -314,6 +320,21 @@ private CommitRange getRangeFromPreviousSuccessfulCommit(EnvVars env, Commit cur
         }
     }
 
+    private EnvVars getEnvVar(FlowExecution execution) throws ExecutionException, InterruptedException, IOException {
+        EnvVars envVars = new EnvVars();
+        if(execution != null) {
+            ListenableFuture<List<StepExecution>> executions = execution.getCurrentExecutions(true);
+            List<StepExecution> stepExecutions = executions.get();
+            for(StepExecution stepExecution : stepExecutions) {
+                EnvVars stepEnvVars = stepExecution.getContext().get(EnvVars.class);
+                if (stepEnvVars != null) {
+                    envVars.putAll(stepEnvVars);
+                }
+            }
```

Comment:
If there are multiple `StepExecution`-s what happens when there's is a var with the same name but a different value? Can they override each other?

Reply:
If innerMostOnly is true you get the most inner one (single one), as far I understood each step should keep state data in different  env var names, but I in case there two vars with same name the last one is kept